### PR TITLE
feat: getters for ctc extradata

### PR DESCRIPTION
**Pull request by tynes**
## Description

Adds getters for the Canonical Transaction Chain extradata. These 2 methods could be bundled together into a single method if that is preferable. This is useful information and also helpful for debugging. I could imagine contracts spanning L1 and L2 wanting to be able to know this information. Right now the only way to get this information is through `ChainStorageContainer.getGlobalMetadata` and then deserializing by hand

### Fixes
- Fixes # [Link to Issue]

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](https://github.com/ethereum-optimism/optimism-monorepo/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol [hunk 1]
```diff
@@ -141,10 +141,40 @@ contract OVM_CanonicalTransactionChain is iOVM_CanonicalTransactionChain, Lib_Ad
             uint40
         )
     {
-        (, uint40 nextQueueIndex,,) = _getBatchExtraData();
```
Comment (tynes) [inline contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol:144]: This change is stylistic to match other places that I saw use no spaces when doing this kind of destructuring

**Comment by tynes** [comment]
@ben-chain I'll rebase this if it is something that we can merge

**Comment by ben-chain** [comment]
Sounds good to me!

**Comment by tynes** [comment]
@ben-chain This has been rebased

**Review by ben-chain — APPROVED** [review]
LGTM!

**Pull request merged**